### ISSUE TITLE
Warning: a promise was created in a handler but was not returned from it

### ISSUE THREAD

**Issue by TitaneBoy** [ortexx/winston-transport-sequelize#1]
Hi there. Thank you for your module. It looks to work great.

But when we use it, it outputs to the console the warning  described in the title: 

`Warning: a promise was created in a handler but was not returned from it`

Can you help me to fix it ? I've tried by my self to make disappear this warning but with no success.

Thank you very much for your support.

**Comment by ortexx** [comment]
Hi!

https://github.com/petkaantonov/bluebird/issues/846

likely, It is old BlueBird bug. Try to update BB to the last version.

**Comment by TitaneBoy** [comment]
Hi. Thank you for your answer.
Technically, this is not a BlueBird issue...But a bad use of "promises" returned by blueBird, if I remember correctly. It happens usually when we do not return correctly a promise. When we use sequelize, it returns us a promise somehow. 

For exemple, when we are making a request with sequelize:

`model.findAll(...).then(function(results){
...})`

The "then" function returns a promise (which is a feature of bluebird). So technically, we must do something like:

`return model.findAll(...)...`

mainly if the model.findAll call is already inside of a `.then(function())` . But we do not must forget that inside the `function()`, we can/must end this function with a `return null`. Then It will not generate the warning.

You can read a good article from bluebird framework about this problem: [Warning promise](http://bluebirdjs.com/docs/warning-explanations.html#warning-a-promise-was-created-in-a-handler-but-none-were-returned-from-it)

I've had already this problem in the past with my code when I was using sequelize. So I'm pretty sure it's not about an old version of bluebird but a bad use of sequelize returns.

As I said before, I've tried to modify your code to make a clean return but I don't understand why it is not working.

Could you help me please ?

Thank you

**Comment by ortexx** [comment]
Have you tried to work with the latest version of bluebird/sequelize? I have no warnings.
In the link above states that BB version <= 3.0.x  considers that the function .emit of EventEmiter is async and it needs a returned value , but it is wrong.  

``` js
 self.model.create(data).then(function (log) {
        self.emit('logged');
        callback(null, log.get());
        // there is BB with bug wants return a value, but there is not necessary.
 })
```

You can return null in code above to check.

**Comment by TitaneBoy** [comment]
Well..my current version of installed bluebird is the v3.4.6 (the latest) and the current version of installed sequelize is the 3.24.3 (the latest version in npm). So I don't think that my versions are that old ;-)

I know that returning null or returning a promise is not very necessary, but this creates a warning in the console.

I don't know what is the IDE you use to develop. I know that if we start node program directly in a console window (for exemple in windows with the CMD shell), no warning is visible. 

Since I am using Visual Studio Code for my development, I can see these warnings if I use badly sequelize and their promises.

**Comment by TitaneBoy** [comment]
Humm...I'm not sure to understand...But I've tried something different. Before, I was calling my first log directly in the server.js of my node server. But I decided to move my first log call somewhere else (in a route function) and now, I don't see any warning. :-/

I will try to investigate later why I see a warning when I'm logging in the server.js file. But for now, it looks to work great, without warnings.

Thank you for your support... :-)
